secure/storage/src/tests/vault.rs

```diff
@@ -99,7 +123,13 @@ fn vault() {
     assert_eq!(storage.get("partial"), Ok(Value::U64(7)));
     assert_eq!(storage.get("full"), Ok(Value::U64(12)));
 
-    // Now begin the generic test
-    storage.reset().unwrap();
-    suite::run_test_suite(Box::new(storage), "VaultStorage");
+    reset_vault_storage(&mut storage);
+    storage
+}
+
+/// Resets the internal state of vault for testing purposes (i.e., clears all stored data).
+fn reset_vault_storage(storage: &mut VaultStorage) {
+    storage
+        .reset()
+        .expect("Failed to reset vault storage state!");
```

Comment:
maybe just call inline storage.rest().unwrap()

Reply:
This doesn't seem like a big thing, but I'd rather have the function call. I think doing the call inline isn't as readable (I try to avoid unwraps!). If it's okay, I'll leave this as it is, and we can see how it fits with the test suite refactor PR that's coming next.

Reply:
If it fails, you're gonna end up looking at the code and callgraph anyway, which makes inlines slightly nicer since you don't jump around the code much to find where it failed.

Reply:
btw, let me click resolved or at least acknowledge so I don't have to jump into things that seem unclear :P